### PR TITLE
Link-detail-search-section. Replaced link spec icon with entity spec icon

### PR DESCRIPTION
**Pull request by ozelenyi**
for the entity's links type "SEARCH" in the list of entities selection
AS IS: the icons shown in the list refers to the icon of the link type or parent entity type (PRICES)
TO BE: icons shown in the list should be taken from the linked entity full type (PRICES.USAGE)

### CODE REVIEW

File: src/app/xm-entity/link-detail-dialog/link-detail-search-section.component.html [hunk 1]
```diff
@@ -30,7 +30,7 @@
         <td>
           <div class="xm-avatar-img-container">
             <img src="{{xmEntity.avatarUrl ? xmEntity.avatarUrl : './assets/img/placeholder.png'}}">
-            <i class="material-icons" *ngIf="linkSpec.icon && !xmEntity.avatarUrl">{{linkSpec.icon}}</i>
+            <i class="material-icons" *ngIf="!xmEntity.avatarUrl">{{getEntitySpec(xmEntity.typeKey).icon}}</i>
```
Comment (andrewkikot) [inline src/app/xm-entity/link-detail-dialog/link-detail-search-section.component.html]: Maybe we should use helper pipe to get icon?

File: src/app/shared/helpers/xm-entity-icon.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Spec } from '../../xm-entity';
+
+@Pipe({name: 'xmEntityIcon'})
+export class XmEntityIconPipe implements PipeTransform {
+
+  public transform(typeKey: string, spec: Spec): string {
+      return spec.types.filter( (t) => t.key === typeKey).shift().icon;
+  }
```
Comment (andrewkikot) [inline src/app/shared/helpers/xm-entity-icon.pipe.ts]: Wouldn`t it fail if filter func do not find coincidence?

File: src/app/shared/shared.module.ts [hunk 3]
```diff
@@ -143,6 +144,7 @@ const MODULES = [
         FileUploadComponent,
         PrivacyAndTermsDialogComponent,
         PIPES,
+        XmEntityIconPipe,
```
Comment (andrewkikot) [inline src/app/shared/shared.module.ts]: maybe move into PIPE const?

File: src/app/shared/helpers/xm-entity-icon.pipe.ts (hunk 2 above)
Comment (ozelenyi) [reply src/app/shared/helpers/xm-entity-icon.pipe.ts]: Agree. How about this `return entitySpec.icon || '';`

**Pull request merged**